### ISSUE TITLE
Clicking on notification doesn't focus application

### ISSUE THREAD

**Issue by gobonja** [squalou/google-chat-linux#30]
Clicking on notification doesn't focus opened application.

This is happening on both plasma and gnome so I don't hinks it is a bug with desktop environment.

Best regards

**Comment by ankurk91** [comment]
This is DE limitation

* Ubuntu 20.04 Gnome = does not work
* Mint Cinnamon = works

I was able to make it work on all DEs but revert because of contextIsolation violation, see commit

https://github.com/ankurk91/google-chat-electron/commit/e595de8e29077804cf5143edd0ccbb45f50c5be0

BTW, I also maintain my own version of "Google chat" here
https://github.com/ankurk91/google-chat-electron

**Comment by squalou** [comment]
You were faster than me.
Also works on XFCE, it's definitively  de DE systray implementation thing.

One way to make it work is to create an empty "libappindicator3.so" and "libappindicator3.so.1" file somewhere where electron looks for them.
It's an ugly tweak that make electron believe it has to work a certain way.

I added them in the package in `/opt/google-chat-linux` but maybe you need to play around. Try to figure out whrere Gnome or Plasma may have those maybe ?

**Comment by gobonja** [comment]
I tried google chat with nativefier and it works properly.

I will poke around to see if I there something from package built with nativefier that can be added to google-chat-linux.

**Comment by ankurk91** [comment]
Here is the responsible code for notification click in nativefier.

https://github.com/jiahaog/nativefier/blob/cf11a71a7c6efd366266fcf39ac6fc49783dd8c7/app/src/preload.ts#L23

nativefier is using old version of electron.
Electron 12 will force developers to use `contextIsolation` which will cause above code not to work.

**Comment by gobonja** [comment]
I see. @ankurk91 Thanks.

btw. What is the difference with google-chat-linux and your fork google-chat-electron?

**Comment by ankurk91** [comment]
@gobonja 
You can install it and see the difference.
You can install my fork with this without loosing any data, since name is different.
I will write main features in readme soon. :)

**Comment by gobonja** [comment]
I did install it and using it in parallel, but except for the different icons and menu I didn't see any other differences.

**Comment by ankurk91** [comment]
Umm, My app 

* Published to snapcraft, automatic updates
* Run at startup
* Dark mode (coming soon)
* Detect offline, and auto connect when internet comes back
* Auto check for updates and notify user.

I will write a feature list in my readme soon

**Comment by squalou** [comment]
By the way, another client still manages the "click" : it's Ferdi.
A good client to keep all your chat apps in one place (in case you have several), and has some nice features.

Sad thing is last time I checked, they used electron 8 (which has reached end of life), no idea what they'll do next.

**Comment by squalou** [comment]
all this being said, I close this issue as I obviously won't do anything more about it ;-)

(feel free to reopen if you have any additional info)
